### PR TITLE
telemetry: update tests to not use EnvoyFilter

### PR DESCRIPTION
**Pull request by howardjohn**
Split out test changes from https://github.com/istio/istio/pull/47892

### CODE REVIEW

**Review by zirain — APPROVED** [review]

File: pkg/test/fakes/gce_metadata_server/main.go [hunk 1]
```diff
@@ -15,7 +15,7 @@
 package main
 
 import (
-	"context"
+	context "context"
```
Comment (hzxuzhonghu) [inline pkg/test/fakes/gce_metadata_server/main.go:18]: revert

File: pilot/pkg/model/telemetry.go [hunk 2]
```diff
@@ -1119,6 +1119,7 @@ func generateSDConfig(class networking.ListenerClass, telemetryConfig telemetryF
 	}
 
 	cfg.MetricExpiryDuration = durationpb.New(1 * time.Hour)
+	cfg.EnableAuditLog = features.StackdriverAuditLog
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply pilot/pkg/model/telemetry.go:1122]: It enables some "audit log" mode, where access logs go to a different stream. I am working to get the feature fully deprecated/removed, but for now to unblock the removal of the EnvoyFilter moving to a short-lived environment variable.

**Review by ericvn — APPROVED** [review]

File: tests/integration/telemetry/api/setup_test.go [hunk 3]
```diff
@@ -49,15 +69,103 @@ spec:
...
-	cfg.Values["meshConfig.accessLogFile"] = "" // disabled accesslog, will enabled by TelemetryAPI later
-	cfg.Values["telemetry.v2.enabled"] = "false"
+	cfg.ControlPlaneValues = `
+meshConfig:
+  accessLogFile: "" # disable from install, we will enable via Telemetry layer
+  defaultConfig:
+    proxyStatsMatcher:
+      inclusionPrefixes:
+      - istio_custom_total
+    extraStatTags:
```
Comment (kyessenov) [inline tests/integration/telemetry/api/setup_test.go:89]: I don't think we need extraStatTags anymore.

File: tests/integration/telemetry/stackdriver/testdata/security_authz_dry_run/server_access_log_allow_no_policy.json.tmpl [hunk 4]
```diff
@@ -19,7 +19,7 @@
       "source_namespace": "{{ .EchoNamespace }}",
       "destination_service_name": "srv",
       "destination_version": "v1",
-      "log_sampled": "true",
+      "log_sampled": "false",
```
Comment (kyessenov) [inline tests/integration/telemetry/stackdriver/testdata/security_authz_dry_run/server_access_log_allow_no_policy.json.tmpl:22]: What's the reason for the change in value here?

File: tests/integration/telemetry/common/stats.go [hunk 5]
```diff
@@ -1,628 +0,0 @@
...
-	return apps.A
-}
-
-func GetTarget() echo.Target {
-	return apps.B
-}
-
-// TestStatsFilter includes common test logic for stats and metadataexchange filters running
-// with nullvm and wasm runtime.
-func TestStatsFilter(t *testing.T, feature features.Feature, expectedBuckets int) {
```
Comment (lei-tang) [inline tests/integration/telemetry/common/stats.go:106]: Why the PR removes the file tests/integration/telemetry/common/stats.go?

Comment (howardjohn) [reply tests/integration/telemetry/common/stats.go:106]: Please see the PR description in https://github.com/istio/istio/pull/47892. We consolidated the 2 test suites for EnvoyFilter and non-Envoyfilter since we only have non-Envoyfilter now.

File: tests/integration/telemetry/stackdriver/testdata/security_authz_dry_run/server_access_log_allow_no_policy.json.tmpl (hunk 4 above)
Comment (howardjohn) [reply tests/integration/telemetry/stackdriver/testdata/security_authz_dry_run/server_access_log_allow_no_policy.json.tmpl:22]: This is a part of the removal of the "access logging policy". I discussed this briefly with you on chat.

I don't see any real purpose for this field/filter. We just set this field, its not _actually_ sampled as far as I can tell -- the `log_sampled=false` requests still go to the real SD backend(!).

If we do want it still, we can add it as an environment var like we did with the AUDIT mode, but it doesn't seem useful. From my (imperfect) research, it has no usage, nor can I come up with even theoretical ideas on how it could be useful.

**Review by lei-tang — APPROVED** [review]
